subsys/bluetooth/mesh/proxy_cli.c

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright (c) 2021 Xiaomi Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/byteorder.h>
+
+#include <net/buf.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/mesh.h>
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_MESH_DEBUG_PROXY)
+#define LOG_MODULE_NAME bt_mesh_proxy_client
+#include "common/log.h"
+
+#include "mesh.h"
+#include "adv.h"
+#include "net.h"
+#include "rpl.h"
+#include "transport.h"
+#include "host/ecc.h"
+#include "prov.h"
+#include "beacon.h"
+#include "foundation.h"
+#include "access.h"
+#include "proxy.h"
+#include "gatt_cli.h"
+#include "proxy_msg.h"
+
+static struct bt_mesh_proxy_server {
+	struct bt_mesh_proxy_role *srv;
+	bool link_opened;
```

Comment:
`link_opened` is duplicate state with `srv != NULL`, and could be replaced by a static inline function:
```c
static inline is_connected(struct bt_mesh_proxy_server *proxy) {
   return proxy->srv != NULL;
}
```

Reply:
link opened when notify has been enabled！

Reply:
Right, I see, sorry!